hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/NameNode.java

```diff
@@ -781,6 +781,10 @@ protected void initialize(Configuration conf) throws IOException {
       }
     }
 
+    if (namesystem.getIsSnapshotTrashRootEnabled()) {
```

Comment:
how about doing this here:
 @Override
    public void startActiveServices() throws IOException {
      try {
        namesystem.startActiveServices();
        startTrashEmptier(getConf());
      } catch (Throwable t) {
        doImmediateShutdown(t);
      }
    }

Just before starting the trashEmptier thread. We don't need to check for Active or standby state here as these should be called in only Active NameNode.

Reply:
good idea